cmake/clang_dev_tools.cmake

```diff
@@ -0,0 +1,88 @@
+# Run a set of LLVM/Clang tools on the code base to maintain high
+# code standards and catch potential security vulnerabilities
+#
+# Tools ran:
+# - clang-format
+# - clang-tidy
+# - include-what-you-use
+
+option(
+  USE_CLANG_FORMAT
+  "Use clang-format if the program is found."
+  OFF
+)
+option(
+  USE_CLANG_TIDY
+  "Use clang-tidy if the program is found."
+  OFF
+)
+option(
+  USE_IWYU
+  "Use IWYU if the program is found."
+  OFF
+)
+
+# Get c++ files across all targets
+file(
+  GLOB_RECURSE
+  ALL_SOURCE_FILES
+  libzeth/**.?pp libzeth/**.tcc
+  prover_server/**.?pp prover_server/**.tcc
+  mpc_tools/**.?pp mpc_tools/**.tcc
+)
+
+# Adding clang-format target if executable is found
+find_program(CLANG_FORMAT "clang-format")
+if(CLANG_FORMAT)
+  message("clang-format found!")
+    if(USE_CLANG_FORMAT)
```

Comment:
Might make sense to put this condition at the top level (so nothing happens if we don't specify `USE_CLANG_FORMAT`, and terminate with an error (rather than warning message) if clang-tidy is not found.